compiler/codegen/expr.cpp

```diff
@@ -4384,20 +4384,13 @@ DEFINE_PRIM(PRIM_LOCAL_CHECK) {
     // arguments are (wide ptr, line, function/file, error string)
     GenRet lhs = call->get(1);
     Symbol* lhsType = lhs.chplType->symbol;
+    bool hasErrorMsg = (call->numActuals() > 3);
+    const char* error = (hasErrorMsg ?
+                         toVarSymbol(toSymExpr(call->get(2))->symbol())->immediate->v_string :
+                         "cannot access remote data in local block");
```

Comment:
"cannot access remote data in local block" is just one of many uses of PRIM_LOCAL_CHECK, rather than being inherent or fundamentally the right default error. Partly for this reason, and partly because I think it'll make these changes less confusing, could you update the code in insertWideReferences that adds PRIM_LOCAL_CHECK to always add the error message argument? IWR should be able to construct the string literal for the default error message once and use it in the existing call sites. Then, in this region of code, you'll always have the error as get(2) and the line and file after that. You'll need to also update the comment on line 4384.
Thanks.

Reply:
Other than the new use I added, "cannot access remote data in local block" is the only other use that we currently have, but I agree that your proposal is more principled and prefer it to what I did.